src/lib/structures/Command.ts

```diff
@@ -142,7 +145,12 @@ export class Command<PreParseReturn = Args, Options extends Command.Options = Co
 	public messagePreParse(message: Message, parameters: string, context: MessageCommand.RunContext): Awaitable<PreParseReturn> {
 		const parser = new Parser(this.strategy);
 		const args = new ArgumentStream(parser.run(this.lexer.run(parameters)));
-		return new Args(message, this as MessageCommand, args, context) as PreParseReturn;
+		return new MessageArgs(message, this as Command, args, context) as PreParseReturn;
+	}
+
+	public chatInputPreParse(interaction: ChatInputCommandInteraction, context: ChatInputCommand.RunContext): Awaitable<PreParseReturn> {
```

Comment:
Needs documentation

Reply:
if you have copilot chat just ask it to document, or you can dump the whole file in bing chat and tell it to document. Or write it yourself but what's the year? 2010? hah

Reply:
Or literally copy paste message one into this